### PR TITLE
Update tests to Junit5

### LINKED ISSUE #148: Use Junit 5
The only thing we have a dependnecy upon that will be kinda goofy for is the `TempDirRule`.
That can be fixed by doing this:
```kotlin
@ExtendWith(TempDirectory::class)
abstract class AbstractPluginTest {

    private lateinit var tempDir: Path

    @BeforeEach
    fun beforeEach(@TempDirectory.TempDir tempDir: Path) {
        this.tempDir = tempDir
    }

    val projectRoot: File
        get() = tempDir.toFile().resolve("plugin-test").apply { mkdirs() }
```

The `TmpDirectory` extension comes from
`testCompile(group = "org.junit-pioneer", name = "junit-pioneer", version = "0.2.2")`.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Closes #148

### CODE REVIEW

File: plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/AbstractPluginTest.kt [hunk 1]
```diff
@@ -3,18 +3,17 @@ package org.jlleitschuh.gradle.ktlint
...
+import org.junit.jupiter.api.io.TempDir
 import java.io.File
 import java.util.Properties
 
 abstract class AbstractPluginTest {
 
-    @get:Rule
-    val temporaryFolder = TemporaryFolder()
+    @TempDir
+    lateinit var temporaryFolder: File
```
Comment (JLLeitschuh) [inline plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/AbstractPluginTest.kt]: Cool! When did they add this to the official API? I thought it was just part of [junit-pioneer](https://github.com/junit-pioneer/junit-pioneer)!

File: plugin/build.gradle.kts [hunk 2]
```diff
@@ -32,10 +32,15 @@ dependencies {
      */
 
     testImplementation(gradleTestKit())
-    testImplementation("junit:junit:${PluginVersions.junit}")
+    testImplementation("org.junit.jupiter:junit-jupiter:${PluginVersions.junit5}")
+    testImplementation("org.assertj:assertj-core:${PluginVersions.assertJ}")
     testImplementation(kotlin("reflect"))
 }
 
+tasks.withType<Test> {
```
Comment (JLLeitschuh) [inline plugin/build.gradle.kts]: This magical incantation is preferred over that one you have there.

```suggestion
tasks.withType<Test>().configureEach {
```

The one you have will eagerly configure the `Test` task. The one in the suggestion will be only be executed when needed.

File: plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/AbstractPluginTest.kt (hunk 1 above)
Comment (Tapchicoma) [reply plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/AbstractPluginTest.kt]: In latest `5.4` release: https://junit.org/junit5/docs/snapshot/release-notes/index.html#release-notes-5.4.0

**Comment by Tapchicoma** [comment]
@JLLeitschuh For me running tests in IDEA fails because of missing `test.properties` file that are generated by Gradle before running tests.

I will address this in next PR for issue #150.

**Pull request merged**